mappings/net/minecraft/client/render/item/ItemRenderer.mapping

```diff
@@ -4,20 +4,41 @@ CLASS net/minecraft/unmapped/C_xjqetogb net/minecraft/client/render/item/ItemRen
 	FIELD f_curtyzhw COMPASS_WITH_GLINT_GUI_MODEL_MULTIPLIER F
 	FIELD f_ddjgfxsn colors Lnet/minecraft/unmapped/C_bvzkhshe;
 	FIELD f_fbqbidyx ITEM_COUNT_BLIT_OFFSET I
+	FIELD f_kwmbfert ENCHANTED_GLINT_ITEM Lnet/minecraft/unmapped/C_ncpywfca;
 	FIELD f_lhkcxhvw GUI_SLOT_CENTER_Y I
 	FIELD f_lwsojpmk textureManager Lnet/minecraft/unmapped/C_rglkduer;
+	FIELD f_pomzspwm COMPASS_FOIL_TEXTURE_SCALE F
 	FIELD f_qjkoxdyv models Lnet/minecraft/unmapped/C_mcmbqsvm;
 	FIELD f_rklqdmcn builtinModelItemRenderer Lnet/minecraft/unmapped/C_apmxdxib;
 	FIELD f_ruqdegyy TRIDENT_INVENTORY Lnet/minecraft/unmapped/C_jcudnsmp;
+	FIELD f_tzvbwrqo ENCHANTED_GLINT_ENTITY Lnet/minecraft/unmapped/C_ncpywfca;
 	FIELD f_uimgkpsz GUI_SLOT_CENTER_X I
 	FIELD f_vpbavmoi SPYGLASS_IN_HAND Lnet/minecraft/unmapped/C_jcudnsmp;
 	FIELD f_xejymrtr TRIDENT_IN_HAND Lnet/minecraft/unmapped/C_jcudnsmp;
 	FIELD f_yibbkkem SPYGLASS_INVENTORY Lnet/minecraft/unmapped/C_jcudnsmp;
+	METHOD <init> (Lnet/minecraft/unmapped/C_ayfeobid;Lnet/minecraft/unmapped/C_rglkduer;Lnet/minecraft/unmapped/C_wosoicga;Lnet/minecraft/unmapped/C_bvzkhshe;Lnet/minecraft/unmapped/C_apmxdxib;)V
+		ARG 2 textureManager
+		ARG 3 bakedModelManager
+		ARG 4 itemColors
+		ARG 5 builtinModelItemRenderer
+	METHOD m_bhoqsheb renderGuiItemDecorations (Lnet/minecraft/unmapped/C_cnszsxvd;Lnet/minecraft/unmapped/C_mavozmpp;Lnet/minecraft/unmapped/C_sddaxwyk;II)V
+		ARG 4 x
+		ARG 5 y
+	METHOD m_brzgbapu renderItemWithOverridesInGui (Lnet/minecraft/unmapped/C_cnszsxvd;Lnet/minecraft/unmapped/C_usxaxydn;Lnet/minecraft/unmapped/C_sddaxwyk;III)V
```

Comment:
what exactly does this method do? I feel like it's a little too long, and if it works in context `renderItemWithGuiOverrides` would be better
this comment applies to all instances of the mapping

Reply:
Your wording absolutely does not work, this order in the naming is very important:

The method takes an `ItemStack` and will select the appropriate `BakedModel` with overrides taken into account, and will then render it with the GUI transformations.

Reply:
sounds good!